### PR TITLE
Cluster cleanup now ignores cluster agent disconnected error

### PR DESCRIPTION
**Pull request by jakefhyde**
### Background
 On error of cleaning up an imported cluster, the `clusterUnavailable` error was expected, however this was because attempting to get the `UserContext` was always trying to create controllers, which would fail if the deletion timestamp was not nil. Now that this is not necessary, the cleanup is able to use a `UserContext` without creating any controllers, and therefore fails when trying to create a cluster role on the unavailable downstream, thus generating a `url.Error` with `Error() == "cluster agent disconnected"`.

### Changes
This PR fixes the above issue by checking for `cluster agent disconnected` instead of `httperror.ClusterUnavailable`.

## Notes
- It may be necessary to hard refresh (F5) to see these changes in effect, as my imported cluster stayed in the sidebar until doing so (after about 30 minutes).

Related Issue: https://github.com/rancher/rancher/issues/35700

### CODE REVIEW

File: pkg/controllers/management/usercontrollers/controller.go [hunk 1]
```diff
@@ -82,13 +82,8 @@ func (c *ClusterLifecycleCleanup) Remove(obj *v3.Cluster) (runtime.Object, error
...
 	if err != nil {
-		apiError, ok := err.(*httperror.APIError)
-		// If it's not an API error give it back
-		if !ok {
-			return nil, err
-		}
-		// If it's anything but clusterUnavailable give it back
-		if apiError.Code != httperror.ClusterUnavailable {
+		// If it's anything but cluster agent disconnected give it back
+		if !strings.Contains(err.Error(), dialer.ErrAgentDisconnected.Error()) {
```
Comment (thedadams) [inline pkg/controllers/management/usercontrollers/controller.go:86]: I think this is a more idiomatic approach.
```suggestion
		if errors.Is(err, dialer.ErrAgentDisconnected) {
```

**Review by thedadams — CHANGES_REQUESTED** [review]

**Comment by cmurphy** [comment]
> however because the controllers are no longer started in this path the api request to create the cleanup cluster role was instead failing with a different error.

Could you explain this more? What does "no longer started in this path" mean, what changed? What's the different error, is that the cluster disconnected error or is it something else we should address?

**Comment by jakefhyde** [comment]
@cmurphy The issue was caused by the change from 
https://github.com/rancher/rancher/blob/26ce1db8dff7cfca12cfafa7198979bdcb346e34/pkg/controllers/management/usercontrollers/controller.go#L101
to
https://github.com/rancher/rancher/blob/bbde38e8cbf1007ba81ce5e68f674948a0033c0d/pkg/controllers/management/usercontrollers/controller.go#L136
because in the former, an attempt to start the controllers is made, but it bailed out with `clusterUnavailable` when it sees the deletion timestamp is not nil when generating the rest config. Should've clarified that starting the controllers actually has nothing to do with it, just the side effect of attempting to start the controllers failing and generating the error that was expected. It's also worth noting that previously, any error besides cluster not found would be discarded and `clusterUnavailable` would be returned instead:
https://github.com/rancher/rancher/blob/26ce1db8dff7cfca12cfafa7198979bdcb346e34/pkg/clustermanager/manager.go#L453-L488
Also yeah, `cluster agent disconnected` is the error as the POST request fails for creating the cluster cleanup role.

**Review by cmurphy — APPROVED** [review]
Thanks for the explanation!

**Review by thedadams — APPROVED** [review]

**Pull request merged**